docs/cookbook/custom-model.rst

```diff
@@ -0,0 +1,158 @@
+How to add a custom model?
+==========================
+
+In some cases you may be needing to add new models to your application in order to cover unique business needs.
+The proccess of extending Sylius with new entities is simple and intuitive.
+
+As an example we will take a **Supplier entity**, which may be really useful for th shop maintenance.
+
+1. Define your needs
+--------------------
+
+A Supplier needs three essential fields: ``name``, ``description`` and ``enabled`` flag.
+
+2. Generate the entity
+----------------------
+
+Sylius provides the `SensioGeneratorBundle <http://symfony.com/doc/current/bundles/SensioGeneratorBundle/index.html>`_,
+that simplifies the process of adding a model.
+
+You need to use such a command in your project directory.
+
+.. code-block:: bash
+
+    $ php app/console generate:doctrine:entity
+
+The generator will ask you for the entity name and fields. See how it should look like to match our assumptions.
+
+.. image:: ../_images/generating_entity.png
+    :align: center
+
+3. Update the database using migrations
+---------------------------------------
+
+Assuming that you have the databease updated to the state before adding the new entity you should run:
```

Comment:
And I'm not a native English speaker but I would turn this phrase into :
`Assuming that your database was up-to-date before adding the new entity, run:`

If a native English speaker could correct me ;-)